### ISSUE TITLE
Filter gists by fileName

### ISSUE THREAD

**Issue by mindrones** [jvcalderon/gist-client#3]
Just an idea, but it just occurred to me that it could be useful to search for gists containing a file with a specific name, for example a meta file that always has the same name, for example "meta.json" for the Svelte REPL (an example here https://gist.github.com/mindrones/1d8a53bf15e8033a62cca847274286aa#file-meta-json).

**Comment by jvcalderon** [comment]
Could be useful. Maybe I could implement it next week. If you have a concrete idea of implementation, a test defining the interface or something else, feel free to share it by pull request. Thank you!

**Comment by mindrones** [comment]
Basically Svelte REPL saves examples as gists and we're trying to retrieve all public examples that people have written.

For example, this example https://svelte.technology/repl?version=2.15.0&gist=1d8a53bf15e8033a62cca847274286aa is based on this gist https://gist.github.com/mindrones/1d8a53bf15e8033a62cca847274286aa.
As you can see, these gists have a `meta.json` file currently just containing the object `{"svelte": true}`, forcing us to search these gists by content, which is proving difficult.

Hence, I'm thinking that we might just change that convention and [call that meta file `svelte.json`](https://github.com/sveltejs/svelte.technology/issues/382#issuecomment-443832871) so that we wouldn't have to fetch and inspect all files in a gist: we would just have to fetch files named `svelte.json` and check if they're Svelte REPL meta files.

An example API could could be:

```js
.getAll({
    filterBy: [
        {hasFileWithName: "svelte.json"}, // tier: resource
    ]
})
```

Being tier `resource` we wouldn't be able to mix with `userName` and `starred` but we can filter the result later on.

Hope this helps, thanks!

**Comment by jvcalderon** [comment]
I just added the functionality in this [commit](https://github.com/jvcalderon/gist-client/commit/80c9f3d1f3b9c2dbf218d7f06fa5667aa0a6ff5f) Its now available in version [1.1.0](https://www.npmjs.com/package/gist-client/v/1.1.0). Now you can do (for example):
```
gistClient.getAll({
      rawContent: true,
      filterBy: [
          {userName: "mindrones"},
          {filename: "data.json5"},
      ]
  })
```
Thank you for your feedback @mindrones

**Comment by mindrones** [comment]
Hi @jvcalderon, I just tried it and works perfectly thanks a lot for implementing it!
